chain/block.go

```diff
@@ -805,7 +805,7 @@ func (b *StatefulBlock) Marshal() ([]byte, error) {
 	p.PackInt64(b.Tmstmp)
 	p.PackUint64(b.Hght)
 
-	p.PackInt(len(b.Txs))
+	p.PackInt(int32(len(b.Txs))) // FIXME: check if int32 is too large for this use case
```

Comment:
It's very unlikely we'll go over max uint16, but it could be close for comfort, so I think we can keep int32. We could certainly switch to using uint32 as this will never be negative (same for the other lengths of things modified in this PR, most of which are also one per tx)